### ISSUE TITLE
Rationalize new project wizard naming 

### ISSUE THREAD

**Issue by briandealwis** [GoogleCloudPlatform/google-cloud-eclipse#1094]
[@branflake2267 suggested](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/1087#issuecomment-266047912) revisiting our new project names:
> Regarding labeling. I might suggest something like moving Maven next to Java, instead of Maven based. Or better yet, only use one shortcut and then specify in the dialog specify which build type. I like it either way, but looking into the future, if you add more build types might get busy.
>
> - Google App Engine Standard Java Project
> - Google App Engine Standard Java Maven Project
> - [Google App Engine Standard Java Gradle|Ant|Blaze Project]

**Comment by elharo** [comment]
Makes sense. There are some cloud branding guidelines we're going to need to follow here.
